backends-velox/src/main/scala/io/glutenproject/backendsapi/velox/IteratorApiImpl.scala

```diff
@@ -178,6 +210,7 @@ class IteratorApiImpl extends IteratorApi with Logging {
     val nativeResultIterator =
       transKernel.createKernelWithBatchIterator(
         rootNode.toProtobuf.toByteArray,
+        new Array[Array[Byte]](0),
```

Comment:
Could you explain a bit what happens here, and is split info needed for final iterator?

Reply:
SplitInfo only exists when stage contain scan, so the final iterator does not need it I think.

`new Array[Array[Byte]](0)` is an empty byte array which represents empty split infos, I know it's looks wired, maybe use a new proto class for array of split infos would be better, but needs change lots codes of substrait conversion. How do you think about that?

Reply:
I'm wondering if split info is also needed for final iterator if ReadRel does not contain LocalFiles. See https://github.com/oap-project/gluten/pull/4177#discussion_r1439213063.

Reply:
scan split info is not needed for final iterator, but stream split info is needed.

Reply:
Does this PR cover this change or some extra change is needed?

Reply:
> Does this PR cover this change or some extra change is needed?

this patch does change stream split info's behavior, it will be extracted from ReadRel as before.

Reply:
> Final iterator does not need split infos, and an empty byte array is used to represent empty split infos.

Maybe better to add some comments as above, thanks.